sdcflows/transform.py

```diff
@@ -732,18 +737,16 @@ def grid_bspline_weights(target_nii, ctrl_nii, dtype="float32"):
         coords[axis] = np.arange(sample_shape[axis], dtype=dtype)
 
         # Calculate the index component of samples w.r.t. B-Spline knots along current axis
+        # Size of locations is L
         locs = nb.affines.apply_affine(target_to_grid, coords.T)[:, axis]
-        knots = np.arange(knots_shape[axis], dtype=dtype)
-
-        distance = np.abs(locs[np.newaxis, ...] - knots[..., np.newaxis])
-        within_support = distance < 2.0
```

Comment:
IIRC this heuristic is what really saved us the time and memory.

Reply:
Makes sense - should we try to investigate how `cubic()` has been internally implemented?

Reply:
Seems to be via fitpack: https://github.com/scipy/scipy/blob/v1.11.1/scipy/interpolate/_bspl.pyx#L466

I think more importantly, the locations evaluated seems to be x * k:

https://github.com/scipy/scipy/blob/cfe80116aaa145061246b8aec0e98248fbefb835/scipy/interpolate/_bsplines.py#L445-L453

I don't fully understand why this is a CSR matrix, it looks dense to me, but I might be missing something.